File: parallelism.py
import multiprocessing as p
import time


def doit():
    ''' check CPU parallelism
    while True:
        pass
    #'''

    time.sleep(1)
    print('並行')

count=0
def main():
    def listener(x):
        global count

        count+=1
        time.sleep(1)
        print(count)


    threads=5
    pool=p.Pool()
    for i in range(threads):
        pool.apply_async(doit, callback=listener)


    # '''
    while threads!=count:
        time.sleep(1)
    #'''
main()

File: parallelism.m
#import <Foundation/Foundation.h>


void doit(void (^cb)(NSString*)) {
    /* check cpu parallelism
    while (true) { }
    //*/
    
    [NSThread sleepForTimeInterval:1];
    NSLog(@"%s", "並行");
    cb(@"bb");
}


int count=0;
int main(int argc, const char * argv[]) {
    auto threads=5;
    NSCondition* lobj=[[NSCondition alloc] init];
    
    typedef void (^Callback)(NSString*);
    Callback listener = ^(NSString* s){
        [lobj lock];
        
        ++count;
        [NSThread sleepForTimeInterval:1];
        NSLog(@"%d", count);
        
        [lobj unlock];
    };
    
    for (auto i=0; i<threads; ++i) {
        NSThread* t = [[NSThread alloc] initWithBlock: ^{
            doit(listener);
        }];
        
        [t start];
    }
    
    while (threads!=count) [NSThread sleepForTimeInterval:1];
}
